client/src/app/site/mediafiles/services/mediafile-repository.service.ts

```diff
@@ -7,49 +7,86 @@ import { User } from '../../../shared/models/users/user';
+import { DataSendService } from 'app/core/services/data-send.service';
+import { HttpService } from 'app/core/services/http.service';
+import { HttpHeaders } from '@angular/common/http';
 
 /**
- * Repository for files
+ * Repository for MediaFiles
  */
 @Injectable({
-    providedIn: 'root'
+    providedIn: 'root',
 })
 export class MediafileRepositoryService extends BaseRepository<ViewMediafile, Mediafile> {
     /**
-     * Consturctor for the file repo
-     * @param DS the DataStore
+     * Constructor for the mediafile repository
+     * @param DS Data store
+     * @param mapperService OpenSlides class mapping service
+     * @param dataSend sending data to the server
+     * @param httpService OpenSlides own http service
      */
-    public constructor(DS: DataStoreService, mapperService: CollectionStringModelMapperService) {
+    public constructor(
+        DS: DataStoreService,
+        mapperService: CollectionStringModelMapperService,
+        private dataSend: DataSendService,
+        private httpService: HttpService,
+    ) {
         super(DS, mapperService, Mediafile, [User]);
     }
 
     /**
-     * Saves a config value.
+     * Alter a given mediaFile
+     * Usually just i.e change the name and the hidden flag.
      *
-     * TODO: used over not-yet-existing detail view
+     * @param file contains the new values
+     * @param viewFile the file that should be updated
      */
     public async update(file: Partial<Mediafile>, viewFile: ViewMediafile): Promise<void> {
-        return null;
+        const updateFile = new Mediafile();
+        updateFile.patchValues(viewFile.mediafile);
+        updateFile.patchValues(file);
+        await this.dataSend.updateModel(updateFile);
     }
 
     /**
-     * Saves a config value.
+     * Deletes the given file from the server
      *
-     * TODO: used over not-yet-existing detail view
+     * @param file the file to delete
      */
     public async delete(file: ViewMediafile): Promise<void> {
-        return null;
+        return await this.dataSend.deleteModel(file.mediafile);
     }
 
     /**
-     * Saves a config value.
+     * Mediafiles are uploaded using FormData objects and (usually) not created locally.
      *
-     * TODO: used over not-yet-existing detail view
+     * @param file a new mediafile
+     * @returns the ID as a promise
      */
     public async create(file: Mediafile): Promise<Identifiable> {
-        return null;
+        return await this.dataSend.createModel(file);
     }
 
+    /**
+     * Uploads a file to the server.
+     * The HttpHeader should be Application/FormData, the empty header will
+     * set the the required boundary automatically
+     *
+     * @param file created UploadData, containing a file
+     * @returns the promise to a new mediafile.
+     */
+    public async uploadFile(file: FormData): Promise<Identifiable> {
+        const restPath = `rest/mediafiles/mediafile/`;
```

Comment:
Shouldn't this be done in `dataSend` service?

Reply:
This does not create a model, so our dataSend service cannot be used (createModel requires a baseModel as parameter)

Reply:
Oh. I thought `MediaFile` is a BaseModel! Especially since it has a model description at [/openslides/mediafiles/models.py](https://github.com/OpenSlides/OpenSlides/blob/master/openslides/mediafiles/models.py) and since it inherits of [`ProjectableBaseModel`](https://github.com/OpenSlides/OpenSlides/blob/master/client/src/app/shared/models/mediafiles/mediafile.ts), which should be a `BaseModel`-Class

Reply:
`Mediafile` is a BaseModel. That's correct. However, `Mediafile` is not what you are creating when you upload a file, not does `Mediafile` contain data. `Mediafile` simply holds the reference to a file on the server. What you upload is a (form)data binary.

Reply:
Ok. I get this. Thats why you upload `FormData`, not the `MediaFile` itself.

So the `FormData` interface is a candidate for #3999?